analyzers/src/SonarAnalyzer.CSharp/Extensions/IMethodSymbolExtensions.cs

```diff
@@ -27,22 +28,25 @@ namespace SonarAnalyzer.Extensions
     internal static class IMethodSymbolExtensions
     {
         // The signature of the Dispose method on IDisposable
-        internal static bool IsDisposeMethod(this IMethodSymbol symbol) =>
+        public static bool IsDisposeMethod(this IMethodSymbol symbol) =>
             symbol.Name.Equals("Dispose") &&
             symbol.Arity == 0 &&
             symbol.Parameters.Length == 0 &&
             symbol.ReturnsVoid &&
             symbol.DeclaredAccessibility == Accessibility.Public;
 
-        internal static bool IsModuleInitializer(this IMethodSymbol methodSymbol) =>
+        public static bool IsModuleInitializer(this IMethodSymbol methodSymbol) =>
             methodSymbol.AnyAttributeDerivesFrom(KnownType.System_Runtime_CompilerServices_ModuleInitializerAttribute);
 
-        internal static bool IsGetTypeCall(this IMethodSymbol invokedMethod) =>
+        public static bool IsGetTypeCall(this IMethodSymbol invokedMethod) =>
             invokedMethod.Name == nameof(Type.GetType)
             && !invokedMethod.IsStatic
             && invokedMethod.ContainingType != null
             && IsObjectOrType(invokedMethod.ContainingType);
 
+        public static SyntaxNode ImplementationSyntax(this IMethodSymbol method) =>
+            (method.PartialImplementationPart ?? method).DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax();
```

Comment:
```suggestion
            (method.PartialImplementationPart ?? method.PartialDefinitionPart).DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax();
```

Reply:
The behavior of this API confused me quite some time ago already: dotnet/roslyn#22598

Reply:
I don't see the benefit here. The current version falls back on whatever the symbol can provide.